docs/python/tutorial-fastapi.md

```diff
@@ -0,0 +1,596 @@
+        items: list[ItemPayload] = []
+        stored_items: dict[str, str] = redis_client.hgetall("item_name_to_id")
+
+        for name, id_str in stored_items.items():
+            item_id: int = int(id_str)
+
+            item_name_str: str | None = redis_client.hget(
+                f"item_id:{item_id}", "item_name"
+            )
+            if item_name_str is not None:
+                item_name: str = item_name_str
+            else:
+                continue  # skip this item if it has no name
+
+            item_quantity_str: str | None = redis_client.hget(
+                f"item_id:{item_id}", "quantity"
+            )
+            if item_quantity_str is not None:
+                item_quantity: int = int(item_quantity_str)
+            else:
+                item_quantity = 0
+
+            items.append(
+                ItemPayload(item_id=item_id, item_name=item_name, quantity=item_quantity)
+            )
+
+        return {"items": items}
+
+
+    # Route to delete a specific item by id but using Redis
+    @app.delete("/items/{item_id}")
+    def delete_item(item_id: int) -> dict[str, str]:
+        if not redis_client.hexists(f"item_id:{item_id}", "item_id"):
+            raise HTTPException(status_code=404, detail="Item not found.")
+        else:
+            item_name: str | None = redis_client.hget(f"item_id:{item_id}", "item_name")
+            redis_client.hdel("item_name_to_id", f"{item_name}")
+            redis_client.hdel(f"item_id:{item_id}", "item_id")
+            return {"result": "Item deleted."}
+
+
+    # Route to remove some quantity of a specific item by id but using Redis
+    @app.delete("/items/{item_id}/{quantity}")
+    def remove_quantity(item_id: int, quantity: int):
+        if not redis_client.hexists(f"item_id:{item_id}", "item_id"):
+            raise HTTPException(status_code=404, detail="Item not found.")
+
+        item_quantity: str | None = redis_client.hget(f"item_id:{item_id}", "quantity")
+
+        # if quantity to be removed is higher or equal to item's quantity, delete the item
+        if item_quantity is None:
+            existing_quantity: int = 0
+        else:
+            existing_quantity: int = int(item_quantity)
+        if existing_quantity <= quantity:
+            item_name: str | None = redis_client.hget(f"item_id:{item_id}", "item_name")
+            redis_client.hdel("item_name_to_id", f"{item_name}")
+            redis_client.hdel(f"item_id:{item_id}", "item_id")
+            return {"result": "Item deleted."}
+        else:
+            redis_client.hincrby(f"item_id:{item_id}", "quantity", -quantity)
+            return {"result": f"{quantity} items removed."}
+    ```
+
+12. Save the file (`kb(workbench.action.files.save)`). The application should automatically reload.
+
+You can now open the `/docs` page again and test the new routes, leveraging the debugger and the debug console to better understand the code execution.
+
+## Set up the data storage
```

Comment:
I'm not sure if we should set this part as optional, but it may be worth calling out that at this point the user has a working version of this app with the base functionality. This section is pretty lengthy and I can see some users not wanting to go through the data storage setup but feel like they need to in order to "complete" the tutorial if that makes sense.